apps/web/components/booking/BookingListItem.tsx

```diff
@@ -927,29 +952,30 @@ const Attendee = (attendeeProps: AttendeeProps & NoShowProps) => {
 
 type GroupedAttendeeProps = {
   attendees: AttendeeProps[];
+  setAttendeeList: Dispatch<SetStateAction<AttendeeList[]>>;
   bookingUid: string;
 };
 
 const GroupedAttendees = (groupedAttendeeProps: GroupedAttendeeProps) => {
-  const { bookingUid } = groupedAttendeeProps;
-  const attendees = groupedAttendeeProps.attendees.map((attendee) => {
-    return {
-      id: attendee.id,
-      email: attendee.email,
-      name: attendee.name,
-      noShow: attendee.noShow || false,
-    };
-  });
+  const { bookingUid, attendees, setAttendeeList } = groupedAttendeeProps;
   const { t } = useLocale();
   const noShowMutation = trpc.viewer.markNoShow.useMutation({
     onSuccess: async (data) => {
+      const newValue = data.attendees;
+      setAttendeeList((old) =>
+        old.map((attendee) =>
```

Comment:
@Praashh 
Right now, I am updating the state in 𝑂(𝑛2) which is not optimized. To improve this, we can sort the newAttendeeList since our global state, attendees, is already sorted based on id. This will allow us to directly make the changes efficiently. For this, I need to modify the markNoShow mutation to return the id as well, along with the email and noShow, which it is currently returning.

![image](https://github.com/user-attachments/assets/f740f89e-65e8-4185-b501-7abb5671991d)
How does this sound to you?

Reply:
feel free to add your solution, i'll happy to check.

Reply:
> feel free to add your solution, i'll happy to check.

You can check it now 👍